ios/RNCAsyncStorage.m

```diff
@@ -16,9 +16,25 @@
 #import "RCTLog.h"
 #import "RCTUtils.h"
 
+typedef NS_ENUM(NSInteger, RCTStorageLocation) {
+  Documents,
+  ApplicationSupport
```

Comment:
@Krizzu Is this what we want? only support `Documents` and `ApplicationSupport` directories?

Reply:
It's been discussed in #13, where `ApplicationSupport` is more safe  to store app-specific data.

Do you have any suggestion if there should be more/less options to choose from?

Reply:
@Krizzu Oh, I scan those things, so now I think we may need to handle two things:
1. Provide options for user to save data. (I suppose support three options, `Documents`,`Library`,`tmp`, https://developer.apple.com/library/archive/documentation/FileManagement/Conceptual/FileSystemProgrammingGuide/FileSystemOverview/FileSystemOverview.html#//apple_ref/doc/uid/TP40010672-CH2-SW1, `tmp` is also meaningful IMO, provide temp storage).
2. Not backed up by iTunes. (For the `Application Support` directory, it would backed up by iTunes and iCloud, so if we use that, we need use `NSURLIsExcludedFromBackupKey` for files we stored includes manifest file.

Reply:
1) not sure about tmp - do you have any use cases where it might be helpful? I think that would just cause confusion as tmp dir can be deleted deleted by OS at any time.
2) maybe we can add a flag to control that? it could be useful to have this file backed up.